### PR TITLE
Add AutoPot

### PR DESCRIPTION
**Pull request by L1shed**
AutoPot is a module that automatically throws healing potions when the player is below a certain health level.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new module called `AutoPot` that automatically throws potions in Minecraft based on player health and potion effects.
  - Added customizable settings for health thresholds and randomized rotations when throwing potions.

### CODE REVIEW

**Comment by TejasLamba2006** [comment]
Hello @L1shed

Did you test the build?

**Comment by L1shed** [comment]
Hi, yes i did, everything worked fine.

File: src/main/java/keystrokesmod/module/impl/combat/AutoPot.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    public void onPreMotion(final PreMotionEvent event) {
+        ticksSinceLastSplash++;
+
+        if (mc.thePlayer.isInWater() || mc.thePlayer.isInLava() || (BlockUtils.getBlock(new BlockPos(mc.thePlayer.posX, mc.thePlayer.posY-1, mc.thePlayer.posZ )) instanceof BlockAir || BlockUtils.getBlock(new BlockPos(mc.thePlayer.posX, mc.thePlayer.posY-1, mc.thePlayer.posZ )) instanceof BlockLadder))
+            ticksSinceCanSplash = 0;
+        else
+            ticksSinceCanSplash++;
+
+        if (switchBack) {
+            PacketUtils.sendPacket(new C07PacketPlayerDigging(C07PacketPlayerDigging.Action.RELEASE_USE_ITEM, BlockPos.ORIGIN, EnumFacing.DOWN));
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/combat/AutoPot.java:58]: It is best not to do packet operations on legit modules.. It's detectable.

**Comment by TejasLamba2006** [comment]
Honestly, I believe that 'Auto Pot' should not be categorized under 'Combat' at all since it is not directly related to PvP. I think it should be placed in the 'Player' category instead.

File: src/main/java/keystrokesmod/module/impl/combat/AutoPot.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                            final String effectName = p.getEffects(itemStack.getMetadata()).get(0).getEffectName();
+
+                            if ((effectName.contains("regeneration") || effectName.contains("heal")) && mc.thePlayer.getHealth() > health.getInput()) {
+                                continue;
+                            } else {
+                                event.setPitch(randomRot.isToggled() ? RandomUtils.nextFloat(85, 90) : 90);
+                                if (!needSplash) {
+                                    needSplash = true;
+                                } else {
+                                    PacketUtils.sendPacket(new C09PacketHeldItemChange(i - 36));
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/combat/AutoPot.java:96]: bad packet warning, raven has no shared slot component

File: src/main/java/keystrokesmod/module/impl/combat/AutoPot.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                                    PacketUtils.sendPacket(new C08PacketPlayerBlockPlacement(itemStack));
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/combat/AutoPot.java:97]: use ```mc.playerController.onPlayerRightClick()``` is better.

**Comment by L1shed** [comment]
Yes, it makes more sense. I do the changes.

File: src/main/java/keystrokesmod/module/impl/combat/AutoPot.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                            }
+                        }
+
+                        if (acceptedPotions.contains(potionID) && !hasPotionIDActive && ticksSinceLastSplash > 20) {
+                            final String effectName = p.getEffects(itemStack.getMetadata()).get(0).getEffectName();
+
+                            if ((effectName.contains("regeneration") || effectName.contains("heal")) && mc.thePlayer.getHealth() > health.getInput()) {
+                                continue;
+                            } else {
+                                event.setPitch(randomRot.isToggled() ? RandomUtils.nextFloat(85, 90) : 90);
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/combat/AutoPot.java:92]: use ```RotationEvent```

**Review by xia-mc — COMMENTED** [review]
it may result a ban.

**Comment by L1shed** [comment]
I followed the recommendations in the review and tested the build. 
It works as intended.

**Comment by TejasLamba2006** [comment]
Please resolve the review that are fixed.

**Comment by L1shed** [comment]
Of course

File: src/main/java/keystrokesmod/module/impl/player/AutoPot.java [hunk 5]
```diff
@@ -46,17 +45,19 @@ public void onDisable() {
     }
 
     @SubscribeEvent
-    public void onPreMotion(final PreMotionEvent event) {
+    public void onPreMotion() {
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/player/AutoPot.java:48]: ```public void onRotation(RotationEvent event) {```

File: src/main/java/keystrokesmod/module/impl/player/AutoPot.java [hunk 6]
```diff
@@ -89,12 +90,13 @@ public void onPreMotion(final PreMotionEvent event) {
                             if ((effectName.contains("regeneration") || effectName.contains("heal")) && mc.thePlayer.getHealth() > health.getInput()) {
                                 continue;
                             } else {
-                                event.setPitch(randomRot.isToggled() ? RandomUtils.nextFloat(85, 90) : 90);
+                                RotationEvent rotationEvent = new RotationEvent(mc.thePlayer.rotationYaw, randomRot.isToggled() ? RandomUtils.nextFloat(85, 90) : 90);
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/player/AutoPot.java:93]: Maybe I wasn't clear.
```event.setPitch(randomRot.isToggled() ? RandomUtils.nextFloat(85, 90) : 90)```

File: src/main/java/keystrokesmod/module/impl/player/AutoPot.java [hunk 7]
```diff
@@ -89,12 +90,13 @@ public void onPreMotion(final PreMotionEvent event) {
...
+                                MinecraftForge.EVENT_BUS.post(rotationEvent);
                                 if (!needSplash) {
                                     needSplash = true;
                                 } else {
-                                    PacketUtils.sendPacket(new C09PacketHeldItemChange(i - 36));
-                                    PacketUtils.sendPacket(new C08PacketPlayerBlockPlacement(itemStack));
+                                    mc.thePlayer.inventory.currentItem = i-36;
+                                    mc.playerController.onPlayerRightClick(mc.thePlayer, mc.theWorld, itemStack, new BlockPos(mc.thePlayer.posX, mc.thePlayer.posY-1, mc.thePlayer.posZ), EnumFacing.UP, new Vec3(mc.thePlayer.posX, mc.thePlayer.posY-1, mc.thePlayer.posZ));
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/player/AutoPot.java:99]: do a raycast ```RotationUtils.rayCast()```

**Review by xia-mc — CHANGES_REQUESTED** [review]
Maybe I wasn't clear..

File: src/main/java/keystrokesmod/module/impl/player/AutoPot.java (hunk 7 above)
Comment (L1shed) [reply src/main/java/keystrokesmod/module/impl/player/AutoPot.java]: what's the point?

**Comment by xia-mc** [comment]
ahhh i will fix it

**Pull request merged**